### PR TITLE
Feat/address service/doma-4706/search by fiasid

### PR DESCRIPTION
**Pull request by AleX83Xpert**
An ability to search houses by fias id.
For now, the plugin will return only houses. Streets and cities will be ignored.

### CODE REVIEW

File: apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+const { AbstractSearchPlugin } = require('./AbstractSearchPlugin')
+const { generateAddressKey } = require('@address-service/domains/common/utils/addressKeyUtils')
+const get = require('lodash/get')
```
Comment (akarjakin) [inline apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js:3]: we se to sort imports

File: apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+const { createOrUpdateAddressWithSource } = require('@address-service/domains/common/utils/services/search/searchServiceUtils')
+const { DadataSuggestionProvider } = require('../../suggest/providers')
+
+const SEPARATOR = ':'
+
+class SearchByFiasId extends AbstractSearchPlugin {
+
+    /**
+     * @param {String} s
```
Comment (akarjakin) [inline apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js:12]: please add description

File: apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+     * @returns {boolean}
+     */
+    isEnabled (s, params) {
+        const [type, fiasId] = s.split(SEPARATOR, 2)
+
+        return type === 'fiasId' && !!fiasId
+    }
+
+    /**
+     * @param {String} s
```
Comment (akarjakin) [inline apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js:23]: please add description

File: apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        if (get(searchResult, ['data', 'house_fias_id']) !== fiasId) {
+            // For now, we want only houses
+            return null
+        }
+
+        const addressKey = generateAddressKey(searchResult)
+
+        return await createOrUpdateAddressWithSource(
+            godContext,
+            {
```
Comment (akarjakin) [inline apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js:48]: please put this object to a variable and make function execution more readable

File: apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js (hunk 1 above)
Comment (AleX83Xpert) [reply apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js:3]: As I know we have no tools to sort imports at backend scripts.

File: apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js (hunk 3 above)
Comment (AleX83Xpert) [reply apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js:23]: Done. Inherited from the base class.

File: apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js (hunk 2 above)
Comment (AleX83Xpert) [reply apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js:12]: Done

File: apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js (hunk 4 above)
Comment (AleX83Xpert) [reply apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js]: Done

File: apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js (hunk 1 above)
Comment (AleX83Xpert) [reply apps/address-service/domains/common/utils/services/search/plugins/SearchByFiasId.js]: But ok, sorted manually alphabetically.

File: apps/address-service/domains/common/utils/services/suggest/providers/DadataSuggestionProvider.js [hunk 5]
```diff
@@ -199,21 +200,48 @@ class DadataSuggestionProvider extends AbstractSuggestionProvider {
...
 
-        const cached = await redis.get(tin)
+    /**
+     * @param {import('ioredis')} cacheClient
+     * @param {string} searchKey
+     * @param {string} apiUrl
+     * @param {number} ttl
+     * @returns {Promise<*|null>}
+     */
+    async callToDadataCached ({ cacheClient, searchKey, apiUrl, ttl = 3600 }) {
```
Comment (sitozzz) [inline apps/address-service/domains/common/utils/services/suggest/providers/DadataSuggestionProvider.js:231]: Maybe move ttl value to local constant like `ORGANIZATION_TIN_CACHE_TTL `

**Review by sitozzz — APPROVED** [review]

File: apps/address-service/domains/common/utils/services/suggest/providers/DadataSuggestionProvider.js (hunk 5 above)
Comment (AleX83Xpert) [reply apps/address-service/domains/common/utils/services/suggest/providers/DadataSuggestionProvider.js]: Done

**Pull request merged**